### PR TITLE
Fact value primitive types

### PR DESCRIPTION
**Pull request by dottorblaster**
Fix primitive types not displayed properly when rendering fact values

## Testing
- Jest tests added
- Stories added for development purposes

### CODE REVIEW

File: assets/js/components/ExecutionResults/CheckResultDetail/CheckResultDetail.stories.jsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+};
+
+export function Default() {
+  return (
+    <CheckResultDetail
+      checkID={checkID}
+      targetID={target1}
+      targetType={targetType}
+      executionData={executionData}
+      clusterHosts={clusterHosts}
```
Comment (arbulu89) [inline assets/js/components/ExecutionResults/CheckResultDetail/CheckResultDetail.stories.jsx:47]: `clusterHosts` is not a `CheckResultDetail` prop. In fact, the `expectations` prop is missing here.
It doesn't really affect the current view, as the `expectations` are only used to get `expectSame` kind of info.

**Review by arbulu89 — APPROVED** [review]
Looks great!

2 things:
- Could you paste a screenshot of the fixed bug? Some scenario where before a scalar value was put as object and now is fixed
- I noticed in storybook that all the ObjectTree elements have the `depth` value within them. First time I notice it. It was this way before?

**Comment by dottorblaster** [comment]
@arbulu89 thanks!

Before: 
image

After:
image

About the depth: I'm investigating :+1:

File: assets/js/components/ExecutionResults/CheckResultDetail/CheckResultDetail.stories.jsx (hunk 1 above)
Comment (dottorblaster) [reply assets/js/components/ExecutionResults/CheckResultDetail/CheckResultDetail.stories.jsx:47]: You're absolutely right, thank you for the heads up :+1:

**Comment by dottorblaster** [comment]
@arbulu89 fixed the `depth` issue using `transientParams`

https://github.com/thoughtbot/fishery#params-that-dont-map-to-the-result-object-transient-params

:+1:

**Comment by arbulu89** [comment]
> @arbulu89 fixed the `depth` issue using `transientParams`
> 
> https://github.com/thoughtbot/fishery#params-that-dont-map-to-the-result-object-transient-params
> 
> +1

But you didn't commit anything :see_no_evil:

**Comment by dottorblaster** [comment]
LOL sorry

File: assets/js/components/ExecutionResults/CheckResultDetail/CheckResultDetail.stories.jsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+export default {
+  title: 'CheckResultDetail',
+  component: CheckResultDetail,
+};
+
+export function Default() {
+  return (
+    <CheckResultDetail
+      checkID={checkID}
```
Comment (EMaksy) [inline assets/js/components/ExecutionResults/CheckResultDetail/CheckResultDetail.stories.jsx:43]: Just a suggestion, we could use also args as a parameter for the Default function, both works :D

**Review by EMaksy — APPROVED** [review]
Nothing much to add, well done. LGTM! :D

File: assets/js/components/ExecutionResults/CheckResultDetail/CheckResultDetail.stories.jsx (hunk 2 above)
Comment (dottorblaster) [reply assets/js/components/ExecutionResults/CheckResultDetail/CheckResultDetail.stories.jsx]: You're right, having the controls available allows us to change stuff arbitrarily. Great tip

image

**Pull request merged**